### PR TITLE
[WS-5132] Added variable support when rewriting requests

### PR DESCRIPTION
**Pull request by andralungu**
Docker played a bunch of tricks on me :) I had to reinstall the quickstart terminal several times to make this work. 

@ddragosd, @cristianconstantin ping.

### CODE REVIEW

File: src/lua/api-gateway/tracking/validator/rewritingRulesValidator.lua [hunk 1]
```diff
@@ -32,6 +32,16 @@ local _M = BaseValidator:new()
 
 local function getBackendFromRewriteRule( rewrite_rule )
     local backend = rewrite_rule.meta or nil
+    local variableManager = ngx.apiGateway.tracking.variableManager
+    if ( variableManager == nil ) then
+        ngx.log(ngx.WARN, "Please initialize RequestVariableManager before calling this method")
```
Comment (ddragosd) [inline src/lua/api-gateway/tracking/validator/rewritingRulesValidator.lua]: if `variableManager` is `nil` this method yields a runtime exception at line 43 bellow.

Comment (ddragosd) [inline src/lua/api-gateway/tracking/validator/rewritingRulesValidator.lua]: if `variableManager` is `nil` this method yields a runtime exception at line 43 bellow.

**Comment by ddragosd** [comment]
Let's rename the PR to something more descriptive and use it in the commit message as well:

> Added variable support when rewriting requests

**Comment by ddragosd** [comment]
👍 after my comments.

**Comment by cristianconstantin** [comment]
Let's also add a new integration test that covers the recent changes.

**Comment by ddragosd** [comment]
@andralungu ping for @constantincristian's comments

**Pull request merged**